eventlet/support/greendns.py

```diff
@@ -178,16 +187,16 @@ def _readlines(self):
 
         Note that this performs disk I/O so can be blocking.
         """
-        lines = []
         try:
-            with open(self.fname, 'rU') as fp:
-                for line in fp:
-                    line = line.strip()
-                    if line and line[0] != '#':
-                        lines.append(line)
+            with open(self.fname, 'rb') as fp:
+                fdata = fp.read()
         except (IOError, OSError):
-            pass
-        return lines
+            return []
+
+        if not isinstance(fdata, str):  # Python 3
```

Comment:
There should really be some function (e.g. `bytes_to_str`) in the `six` package to deal with this problem, but I couldn't find anything suitable. So there are two solutions - check the version directly, or read in text mode.